src/main/java/frc/robot/subsystems/vision/PhotonVisionIO.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.math.geometry.Transform3d;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+
+public class PhotonVisionIO implements VisionIO {
+
+    private final PhotonCamera camera;
+    private final PhotonPoseEstimator estimator;
+    private Result result;
+
+    public PhotonVisionIO(PhotonCamera camera, Transform3d robotToCamera) {
+        this.camera = camera;
+
+        try {
+            estimator = new PhotonPoseEstimator(
+                    AprilTagFieldLayout.loadFromResource("2023field.fmap"),
```

Comment:
No need to save the file yourself (will it be deployed to the robot anyway? It's not stored in the `deploy` directory).
You can use the `AprilTagFields` enum instead, see an [example](https://docs.photonvision.org/en/latest/docs/programming/photonlib/robot-pose-estimator.html#creating-an-apriltagfieldlayout) here.